### ISSUE TITLE
Full cargo in dedicated salvage

### ISSUE THREAD

**Issue by sigmastk** [ISeeDEDPpl/Questor#169]
Hi,

I am using latest directeve.dll (dload it today from support site) and latest ISeeDEDPpl.

I have noticed strange bug when using dedicated salvager. When ship reaches limit defined in xml (default is 100m3) it stops looting, but continue to salvage wrecks, until it reaches 100% of cargospace. Then it goes and unload loot.

I have tried to define other valued than 100 and it behaves similar when reach limit. If I put 0, then sometimes it stops working at all when it has less than 1 m3.

I am wondering, what kind of logs would be useful to post them here ?

**Comment by I3Questor** [comment]
Yeah, I have this exact same issue.  I thought it was just me.

**Comment by ISeeDEDPpl** [comment]
This should now be fixed in current experimental
https://github.com/ISeeDEDPpl/Questor/commit/73a36bea08112284e220342856a2bc61134794d4

**Comment by sigmastk** [comment]
Confirming that it is working now:) 

Thanks for all the great work you put in q ! :)

**Comment by sigmastk** [comment]
Hmm, after further testing, I discovered that after questor recognizes that cargo is indeed full, it stuck in space.

With my ninja c# skills (read, as a trained monkey riding a roller-skates :D )  I discovered that maybe it's a "copy/paste" error.

In file DedicatedBookmarkSalvagerBehavior.cs on line 608 there is line: 
**_States.CurrentCombatMissionBehaviorState = CombatMissionsBehaviorState.GotoBase;** .

I replaced that line with 
**_States.CurrentDedicatedBookmarkSalvagerBehaviorState = DedicatedBookmarkSalvagerBehaviorState.GotoBase;** 

And now it works as intended.

Maybe it would be ok to replace that in original file on github ?
